src/backends/perf_pt/decode.c

```diff
@@ -325,6 +348,36 @@ handle_events(struct pt_block_decoder *decoder, int *decoder_status, struct perf
     return ret;
 }
 
+/*
+ * Decides if a block is terminated by a control flow dispatch.
+ *
+ * This is used to decide if libipt gave us a partial block or not.
+ */
+static bool
+block_is_terminated(struct pt_block *blk)
+{
+    bool ret;
```

Comment:
This seems a tedious way of just saying `return true` below?

Reply:
I started out with early returns, but you end up with unreachable code and another panic.

What do you prefer?

Reply:
Ah, one of those :/ OK, leave as-is.

Reply:
Roger.